Sources/PointFree/Stripe.swift

```diff
@@ -333,15 +359,18 @@ private func createCustomer(user: Database.User, token: Stripe.Token.Id, vatNumb
 private func createSubscription(
   customer: Stripe.Customer.Id,
   plan: Stripe.Plan.Id,
-  quantity: Int
+  quantity: Int,
+  coupon: SubscribeData.Coupon?
   )
   -> EitherIO<Error, Stripe.Subscription> {
 
-    return stripeDataTask("subscriptions?expand[]=customer", .post([
-      "customer": customer.rawValue,
-      "items[0][plan]": plan.rawValue,
-      "items[0][quantity]": String(quantity),
-      ]))
+    var params: [String: Any] = [:]
+    params["customer"] = customer.rawValue
+    params["items[0][plan]"] = plan.rawValue
+    params["items[0][quantity]"] = String(quantity)
+    params["coupon"] = coupon?.rawValue
```

Comment:
broke this out into multiple lines cause I dont think we can put nil values directly into a `[String: Any]` initialization. have to use subscripts...

Reply:
Ah, I think we just use `[String: String]`, so we could tighten that up maybe and use `compactMapValues`.